packages/theme-check-common/src/index.ts

```diff
@@ -157,7 +157,7 @@ function createContext<T extends SourceCodeType, S extends Schema>(
         end: getPosition(file.source, problem.endIndex),
         fix: problem.fix,
         suggest: problem.suggest,
-      } as Offense<T>);
+      } as Offense<T> as Offense);
```

Comment:
Compiler got confused and crashed here without that extra `as Offense` since `offenses` is `Offense[]`

`Offense<T>` for some reason stopped collapsing to `Offense` in 5

Reply:
Could we simplify to this then?

```suggestion
      } as Offense);
```